atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java

```diff
@@ -64,12 +65,13 @@ protected final <T, E extends Exception> T runTaskThrowOnConflict(TransactionTas
     protected final <T, E extends Exception> T runTaskThrowOnConflictWithCallback(
             TransactionTask<T, E> task, CallbackAwareTransaction txn, Runnable callback)
```

Comment:
This is yet another "callback" how is this used, do we know?

Reply:
This seems to be used only for `PreCommitCondition#cleanup` callbacks. It's a bit weird as these are likely only unlocks (which again should be registered as part of `closer#close`. They now are, albeit in a convoluted way (via `onCommitOrAbort` handle).